packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,266 @@
+        """
+        Connect.
+
+        :return: an asynchronous queue, that constitutes the communication channel.
+        """
+        with self.lock:
+            if self.stopped:
+                self._httpCall = HTTPCalls(
+                    server_address=self.provider_addr, port=self.provider_port
+                )
+                self.stopped = False
+                self.thread.start()
+                logger.debug("HTTP Channel is connected.")
+                self.try_register()
+
+    def send(self, envelope: Envelope) -> None:
+        """
+        Process the envelopes.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        assert self._httpCall is not None
+
+        if self.excluded_protocols is not None:
+            if envelope.protocol_id in self.excluded_protocols:
+                logger.error(
+                    "This envelope cannot be sent with the oef connection: protocol_id={}".format(
+                        envelope.protocol_id
+                    )
+                )
+                raise ValueError("Cannot send message.")
+
+        self._httpCall.send_message(
+            sender_address=envelope.sender,
+            receiver_address=envelope.to,
+            protocol=str(envelope.protocol_id),
+            context=b"None",
+            payload=envelope.message,
+        )
+
+    def receiving_loop(self) -> None:
+        """Receive the messages from the provider."""
+        assert self._httpCall is not None
+        assert self.in_queue is not None
+        assert self.loop is not None
+        while not self.stopped:
+            messages = self._httpCall.get_messages(
+                sender_address=self.address
+            )  # type: List[Dict[str, Any]]
+            for message in messages:
+                logger.debug("Received message: {}".format(message))
+                envelope = Envelope(
+                    to=message["TO"]["RECEIVER_ADDRESS"],
+                    sender=message["FROM"]["SENDER_ADDRESS"],
+                    protocol_id=PublicId.from_str(message["PROTOCOL"]),
+                    message=message["PAYLOAD"],
+                )
+                self.loop.call_soon_threadsafe(self.in_queue.put_nowait, envelope)
+            time.sleep(0.5)
+        logger.debug("Receiving loop stopped.")
+
+    def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        :return: None
+        """
+        assert self._httpCall is not None
+        with self.lock:
+            if not self.stopped:
+                self._httpCall.unregister(self.address)
+                # self._httpCall.disconnect()
+                self.stopped = True
+                self.thread.join()
+
+
+class HTTPConnection(Connection):
+    """Proxy to the functionality of the RESTful API."""
+
+    def __init__(self,
+                 address: Address,
+                 api_spec: str,  # In JSON serialized form, might switch to yaml or server_stub later
```

Comment:
The swagger spec above is in yaml, so why not go with a path to that file: `api_spec_path:`